hnn_core/network_builder.py

```diff
@@ -377,78 +385,59 @@ def _create_cells_and_feeds(self):
+                    cell = BasketCell(pos=src_pos, gid=gid)
                 self.cells.append(cell)
 
             # external inputs are special types of artificial-cells
             # 'common': all cells impacted with identical TIMING of spike
-            # events. NB: cell types can still have different weights for how
-            # such 'common' spikes influence them
-            elif src_type == 'common':
-                # print('cell_type',cell_type)
-                # to find param index, take difference between REAL gid
-                # here and gid start point of the items
-                p_ind = gid - self.net.gid_dict['common'][0]
-
-                # new ExtFeed: target cell type irrelevant (None) since input
-                # timing will be identical for all cells
-                feed = ExtFeed(feed_type=src_type,
-                               target_cell_type=None,
-                               params=self.net.p_common[p_ind],
-                               gid=gid)
-                feed_cell = _ArtificialCell(feed.event_times,
-                                            params['threshold'])
+            # events. NB: cell types can still have different weights for
+            # how such 'common' spikes influence them
+            else:
+                # XXX figure out how to index the Nth spike in each list
+                gid_idx = gid - self.net.gid_dict[src_type][0]
+                et = this_trial_event_times[src_type][gid_idx]
+                feed_cell = _ArtificialCell(et, params['threshold'],
+                                            gid=gid)
                 self._feed_cells.append(feed_cell)
 
-            # external inputs can also be Poisson- or Gaussian-
-            # distributed, or 'evoked' inputs (proximal or distal)
-            # these are cell-specific ('unique')
-            # XXX src_type is 'common', 'evprox1', 'evdist1', 'evgauss', etc.
-            elif src_type in self.net.p_unique.keys():
-                gid_target = gid - self.net.gid_dict[src_type][0]
-                target_cell_type = self.net.gid_to_type(gid_target)
-
-                # new ExtFeed, where now both feed type and target cell type
-                # specified because these feeds have cell-specific
-                # CONNECTION parameters, i.e., AMPA and NMDA weights
-                feed = ExtFeed(feed_type=src_type,
-                               target_cell_type=target_cell_type,
-                               params=self.net.p_unique[src_type],
-                               gid=gid)
-                feed_cell = _ArtificialCell(feed.event_times,
-                                            params['threshold'])
-                self._feed_cells.append(feed_cell)
-            else:
-                raise ValueError('No parameters specified for external feed '
-                                 'type: %s' % src_type)
+    def _assign_cells_and_feeds_to_neuron_network(self):
```

Comment:
how about:

```suggestion
    def _gid_to_pc(self):
```

descriptive enough? I don't want the name to be too vague ...

also maybe add an argument called `threshold` so we don't accidentally add more dependencies on `params` inside the function.

Reply:
+1 on name for now, might change once it becomes clearer how to implement drives later.
added the param, as it's trivially easy here (much harder in `_create_cells...`)